### DISCUSSION TITLE
Using multiple stores for same components

### DISCUSSION THREAD

**Discussion by tonymaynard97** [pmndrs/zustand#1568] [Q&A]
I have some charts across different pages, which I need to persist the state for so that the settings are reloaded when the user clicks on a particular page. For example, the user is on page A chart 1 and they change the zoom to 2x, I don't want to apply this to every single chart and when they click back to page 1 I want to reinstate the zoom level. 

I'm new to zustand but is there a way to do this? I took a look at the context wrapper section but I believe this is getting deprecated from[ the documentation](https://docs.pmnd.rs/zustand/previous-versions/zustand-v3-create-context).  I'm not sure this would even work anyway as the store would get recreated on mount. Is there a way to handle these with zustand? The only other way I can think of is having a nested keyed object for each chart.

**Comment by dai-shi** [comment]
Please see https://github.com/pmndrs/zustand/discussions/1180 about how to use store with context.

I'm not sure if context fits with your use case.

> The only other way I can think of is having a nested keyed object for each chart.

Sounds good too.

**Reply by tonymaynard97** [reply]
Great thanks, if I was going down the keyed object root, do you have an example of updating arbitrary keys if they maybe don't exist?

**Reply by dai-shi** [reply]
Is it a JS question? Can you explain it with a code snippet?
